### PR TITLE
Gat

### PR DESCRIPTION
**Pull request by yunjinli**

### CODE REVIEW

File: crowd_nav/configs/env.config [hunk 1]
```diff
@@ -18,14 +18,14 @@ discomfort_penalty_factor = 0.5
...
-human_num = 5
+human_num = 10
 static_obstacle_num = 10
 min_obst_offset = 1
-obstacle_max_radius = 2.5
-obstacle_min_radius = 0.5
-boundary = 20
+obstacle_max_radius = 0.3
+obstacle_min_radius = 0.3
+boundary = 15
```
Comment (kkaytekin) [inline crowd_nav/configs/env.config:28]: todo @kkaytekin : change the default configs

File: crowd_nav/args.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def __init__(self,mode='train'):
             self.parser.add_argument('--plot_cr', default=False, action='store_true')
             self.parser.add_argument('--plot_time', default=False, action='store_true')
             self.parser.add_argument('--plot_reward', default=True, action='store_true')
-            self.parser.add_argument('--plot_train', default=True, action='store_true')
+            self.parser.add_argument('--plot_train', default=False, action='store_true')
```
Comment (kkaytekin) [inline crowd_nav/args.py]: todo @kkaytekin : set default to true

File: README.md [hunk 3]
```diff
@@ -1,29 +1,10 @@
 # CrowdNav
```
Comment (kkaytekin) [inline README.md:1]: todo @kkaytekin : add cl instructions

File: crowd_nav/configs/train.config [hunk 4]
```diff
@@ -16,14 +16,14 @@ rl_learning_rate = 0.001
 # number of batches to train at the end of training episode
 train_batches = 100
 # training episodes in outer loop
-train_episodes = 1000
+train_episodes = 10000
```
Comment (kkaytekin) [inline crowd_nav/configs/train.config:19]: maybe keep this at 3k so that its comparable to sarl?

File: crowd_nav/general_evaluation.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+import numpy as np
+import gym
+from crowd_nav.utils.explorer import Explorer
+from crowd_nav.policy.policy_factory import policy_factory
+from crowd_sim.envs.utils.robot import Robot
+from crowd_sim.envs.policy.orca import ORCA
+from crowd_nav.args import Parser
+import sys
+
+
```
Comment (kkaytekin) [inline crowd_nav/general_evaluation.py:15]: maybe add a comment about purpose of this file?
something like:
`#Script to serialize testing`

File: crowd_nav/policy/gat4sn.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        Wh = Wh.view(B, N, -1)
+        assert Wh.shape == (B, N, self.out_dim)
+        robot_state = Wh[:, 0, :]
+        assert robot_state.shape == (B, self.out_dim)
+        other_agent_states = Wh[:, 1:, :]
+        assert other_agent_states.shape == (B, N - 1, self.out_dim)
+        cat1 = torch.cat((robot_state.view(B, 1, -1).expand(B, N - 1, -1), other_agent_states), dim = -1)
+        assert cat1.shape == (B, N - 1, 2 * self.out_dim)
+        attention = torch.mm(cat1.view(B * (N - 1), -1), self.a).squeeze()
+        attention = attention.view(B, N - 1)
```
Comment (kkaytekin) [inline crowd_nav/policy/gat4sn.py:45]: maybe comment out the `assert` statements?

File: crowd_nav/policy/gat4sn.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        '''
+        Dimension of state = B x N x D
+        B = batch_size
+        N = num_humans + num_obstacles
+        D = robot_state_dim + other_agent_states_dim
+        '''
+        B, N, D = state.shape
+
+        robot_state = state[:, 0, :self.self_state_dim]
+        assert robot_state.shape == (B, self.self_state_dim)
```
Comment (kkaytekin) [inline crowd_nav/policy/gat4sn.py]: maybe comment out the `assert` statements?

File: crowd_nav/utils/plot.py [hunk 8]
```diff
@@ -15,7 +15,10 @@ def main():
 
     # define the names of the models you want to plot and the longest episodes you want to show
     #models = ['LSTM-RL', 'SARL', 'OM-SARL']
-    models = ['Continuing', 'Slowing', 'Expanding','Expanding & Moving']
+    # models = ['Continuing', 'Slowing', 'Expanding','Expanding & Moving']
+    # models = ['GAT4SN', 'SARL']
```
Comment (kkaytekin) [inline crowd_nav/utils/plot.py]: todo @kkaytekin : clear out the commented lines

File: crowd_sim/envs/crowd_sim.py [hunk 9]
```diff
@@ -389,9 +391,21 @@ def reset(self, phase='test', test_case=None):
             ## The random seed should also be added here, otherwise the
             ## generated environment would be totally different
             np.random.seed(counter_offset[phase] + self.case_counter[phase])
-            self.robot_gx, self.robot_gy = self.generate_agent_goal()
+            # self.robot_gx, self.robot_gy = self.generate_agent_goal(goal_range = self.square_width / 2)
             # self.robot.set(0, -self.circle_radius, 0, self.circle_radius, 0, 0, np.pi / 2)
-            self.robot.set(-self.robot_gx, -self.robot_gy, self.robot_gx, self.robot_gy, 0, 0, np.pi / 2)
+            # self.robot_gx, self.robot_gy = self.generate_agent_goal(goal_range = self.square_width / 2)
```
Comment (kkaytekin) [inline crowd_sim/envs/crowd_sim.py]: todo: clear the commented lines

File: crowd_sim/envs/crowd_sim.py [hunk 10]
```diff
@@ -649,6 +678,11 @@ def step(self, action, update=True):
             done = False
             info = Danger(dmin)
         else:
+            # delta_d = (-(norm(end_position - np.array(self.robot.get_goal_position()))) + (norm(np.array([robot_x, robot_y]) - np.array(self.robot.get_goal_position()))))
```
Comment (kkaytekin) [inline crowd_sim/envs/crowd_sim.py]: should we remove this penalty statement all together? or allow it through a config file?

File: crowd_sim/envs/crowd_sim.py [hunk 11]
```diff
@@ -967,11 +1039,17 @@ def plot_value_heatmap():
                 # when any key is pressed draw the action value plot
                 fig, axis = plt.subplots()
                 speeds = [0] + self.robot.policy.speeds
-                rotations = self.robot.policy.rotations + [np.pi * 2]
+                # rotations = self.robot.policy.rotations + [np.pi * 2]
```
Comment (kkaytekin) [inline crowd_sim/envs/crowd_sim.py]: todo: clear comments

**Review by kkaytekin — APPROVED** [review]
implementation-wise all fine, suggested removing some comments

File: crowd_nav/configs/train.config (hunk 4 above)
Comment (yunjinli) [reply crowd_nav/configs/train.config:19]: Really? But why is it 3k? I thought we always use 10k. I guess 3k is for imitation learning

File: crowd_sim/envs/crowd_sim.py (hunk 10 above)
Comment (yunjinli) [reply crowd_sim/envs/crowd_sim.py]: I'll just remove it. According to my experiment, this penalty is not helpful.

File: crowd_sim/envs/crowd_sim.py (hunk 11 above)
Comment (yunjinli) [reply crowd_sim/envs/crowd_sim.py]: noted

File: crowd_nav/configs/train.config (hunk 4 above)
Comment (kkaytekin) [reply crowd_nav/configs/train.config:19]: oh i must have lost focus there. yes, 3k is for imitation learning. nothing wrong here.

**Pull request merged**